src/pint/models/model_builder.py

```diff
@@ -532,6 +565,71 @@ def choose_model(self, param_inpar):
                 selected_cates[cate] = cp
         return selected_components, conflict_components, param_not_in_pint
 
+    def choose_binary_model(self, param_inpar, force_binary_model=None, allow_T2=False):
+        """Choose the BINARY model based on the parfile.
+
+        Parameters
+        ----------
+        param_inpar: dict
+            Dictionary of the unique parameters in .par file with the key is the
+            parfile line. :func:`parse_parfile` returns this dictionary.
+
+        force_binary_model : str
+            When set to some binary model, this will override the binary model
+            set in the parfile.
+
+        allow_T2 : True, False, or str optional
+            Whether to convert a T2 binary model to an appropriate underlying
+            binary model. Default is False, and will throw an error upon
+            encountering the T2 binary model. If True, the binary model will be
+            converted to the most appropriate PINT-compatible binary model.
+
+        Returns
+        -------
+        str
+            Name of the binary component
+
+        Note
+        ----
+        If the binary model does not have a PINT model (e.g. the T2 model), an
+        error is thrown with the suggested model that could replace it. If an
+        appropriate model cannot be found, no suggestion is given
```

Comment:
Some time ago, I made a PR (#1701) to give tailored error messages for un-implemented binary models like MSS. How does this PR interact with that?

Reply:
Please also mention the allow_T2 option.